core/src/main/java/com/flipkart/gjex/core/healthcheck/RotationManagementBasedHealthCheck.java

```diff
@@ -0,0 +1,49 @@
+package com.flipkart.gjex.core.healthcheck;
+
+import com.flipkart.gjex.core.logging.Logging;
+import io.dropwizard.metrics5.health.HealthCheck;
+
+import javax.inject.Singleton;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+/**
+ * Rotation management based health check for the app
+ * @author ajay.jalgaonkar
+ */
+
+@Singleton
+public class RotationManagementBasedHealthCheck extends HealthCheck implements Logging {
+  private static final String BIR = "bir";
+  private static final String OOR = "oor";
+  private static final AtomicBoolean rotationStatus = new AtomicBoolean(true);
```

Comment:
It will be better to start instance as oor and make is bir once all startup processes are completed .

Reply:
Lets not change the default behaviour, as this would be breaking change. If this is required, it can be done via configuration imo.

Reply:
Now binding of RotationManagementBasedHealthCheck is done in ApiModule